lib/sales_engine.rb

```diff
@@ -30,7 +32,9 @@ def self.from_csv(data)
   end
 
   def get_data
-    @merchant_data = merchants.get_revenue
+    @transaction_data   = transactions.successful_transactions
+    @invoice_item_data  = invoice_items.total(@transaction_data.keys)
+    @merchant_data      = merchants.get_revenue
   end
```

Comment:
maybe just break these out into methods?
def transaction_data
    transactions.successful_transactions
 end

etc

Reply:
the code thing on github is surprisingly horrible

Reply:
If I did that, they would go through all their enumerables again =/

Reply:
what if they were memoized?

Reply:
I'll give it a shot =)